src/http/server/handlers/static_file_handler.cr

```diff
@@ -46,18 +46,23 @@ class HTTP::StaticFileHandler
       return
     end
 
-    expanded_path = File.expand_path(request_path, "/")
-    if is_dir_path && !expanded_path.ends_with? "/"
-      expanded_path = "#{expanded_path}/"
+    request_path = Path.posix(request_path)
+    expanded_path = request_path.expand("/")
+    if is_dir_path && !expanded_path.ends_with_separator?
+      expanded_path = expanded_path.join("")
```

Comment:
WDYT about adding `Path#append_separator`?

Reply:
It seems a bit hard to name, `append_separator` is a bit imperative.

Reply:
Perhaps just encoding dir-ness by whether the path ends in `/` is the problem. Could we extract thta information early into a boolean and operate on it seperately to the path?

Reply:
There is already `is_dir_path` for that. But frankly, I don't understand everything that goes on with that. `is_dir_path && !expanded_path.ends_with_separator?` doesn't seem it could ever be true, because it's equivalent to:
```cr
original_path.ends_with?("/") && !File.expand_path(request_path(URI.decode(original_path)), "/").ends_with?("/")
```
Neither of the applied operations `URI.decode`, `#request_path` (no-op), `File.expand_path` could (either alone or in combination) result in `/` being removed at the end of the string. Thus, this expressions seems to be a clear contradiction. This code was added in #2544. 

I'd suggest to simply remove these lines as they don't seem to have any purpose. Specs are not affected.
Maybe @MakeNowJust has some insights, though.

Reply:
On that note, I think the entire `#call` method could need some refactoring. I'd rather move that to a separate PR, though. It's not essential.

Reply:
When I write this, `Fie.expand_path` removes the final path separator, so this line is needed. However now is not, so we can remove simply. Thanks @straight-shoota and @RX14

Reply:
Great that you can still recall the circumstances.